src/ArmTemplates/Common/Templates/Abstractions/TemplateResource.cs

```diff
@@ -23,5 +23,12 @@ public class TemplateResource
 
         [JsonProperty(Order = -2)]
         public string[] DependsOn { get; set; }
+
+        [JsonIgnore]
+        public string OriginalName { get; set; }
```

Comment:
I would add xml comments for properties, describing the intention. Also, please, go through all specific `TemplateResource` classes and delete former `OriginalName` property, so that we don't leave some properties unused.

Reply:
Those attributes removed from the TemplateResource and added to the specific templates, during moving them to the TemplateResource model with making generic way of processing incoming data will mentioned things in the comment